### DISCUSSION TITLE
Using Recent VC (CL.exe) in Command-Environment Builds with RayLib

### DISCUSSION THREAD

**Discussion by orcmid** [raysan5/raylib#2432] [Ideas]
Since around Visual Studio 2017 (VS 15.0) the common installer for Visual Studio and Visual Studio Build Tools no longer provides environment variables that are usable in locating the installed `vcvarssall.bat` script for setting up the build environment.  There is no easy means of automatic discovery.

The recommended alternative is to perform one of the installed shortcuts for build environment that are placed in a start page, such as the one in "All Programs" for a VC x64-building command environment.
![VSBuildTools-x64NativeCommandPrompt](https://user-images.githubusercontent.com/3372580/162793422-56c51803-db13-491c-a663-040fc5de0126.png).

Using that shortcut provides for sessions like this in a simple Hello World confirmation.  
![HelloWorld](https://user-images.githubusercontent.com/3372580/162794381-1b743c3c-56d1-4f1b-82fb-392baa3580ef.png)

There is a bit more involved in finding the prompt and also making a desktop shortcut that uses it to start the command environment where you want, and with other command prompt option that the user might want.  There are extra steps for customizing a developer setup that is easy to initiate.

With these shortcuts, the `raylib/projects/scripts/build-windows.bat` script will find the necessary tools when used within the command environment.

There is a giant bonus to this method.  To use Visual Studio Code, it should be launched within a command environment established this way.  Just use the command `code`.  Visual Studio Code recognizes the established environment and provides intellisense for code selected by `#include` of  libraries in the environment.   There is a way to set a `.vscode` workspace so the raylib header files are also treated this way.

I have not found an easy way to introduce all of this as tips in `raysan5/raylib`.  I'd like some advice.

**Comment by orcmid** [comment]
I've gone ahead and created a kit, [VCrayApp](https://orcmid.github.io/nfoTools/dev/D211101/) for making raylib apps using the Microsoft VC/C++ compiler and the Windows command line (CMD) for compiling, running, and testing.

I have now provided a [VCrayApp-0.0.0](https://orcmid.github.io/nfoTools/dev/D211101/D211101d/) alpha release.  It is definitely alpha in the sense that supporting documentation and tutorial are slim or absent and will be developed before a beta release.  In particular, testing and confirmation with the examples for different raylib versions have been barely started.  This will change with the beta release, which should be mostly about accompanying documentation for beginners and, ideally, the final successful release candidate.

**Comment by raysan5** [comment]
@orcmid Great work! Thank you very much! I'm adding it to the raylib Wiki for raylib CL building on Windows: https://github.com/raysan5/raylib/wiki/Working-on-Windows#building-raylib-project-with-vc-clexe-in-command-line-environment
